index.ts

```diff
@@ -259,14 +256,41 @@ export namespace Foreign {
     fungibleVotePercent: BigInteger;
   }
 
-  export interface ContractEnv;
+  export interface TallyStateDatum {
+    proposal: ProposalType;
+    proposalEndTime: POSIXTime;
+    for: BigInteger;
+    against: BigInteger;
+  }
 
-  export interface ServerConfig {
-    port: number;
-    host: string;
-    secure: boolean;
-    path: Maybe<string>;
+  export declare class ProposalType {}
+  export declare class Upgrade extends ProposalType {
+    value0: CurrencySymbol;
+  }
+  export declare class General extends ProposalType {
+    value0: Address;
+    value1: BigInteger;
   }
+  export declare class Trip extends ProposalType {
+    value0: Address;
+    value1: Address;
+    value2: BigInteger;
+  }
+
+  export interface VoteDatum {
+    proposalTokenName: TokenName;
+    direction: VoteDirection;
+    voteOwner: Address;
+    returnAda: BigInteger;
+  }
+
+  export declare class VoteDirection {};
+  export declare class For extends VoteDirection {};
```

Comment:
I'm always skeptical about using anything similar to keywords in any language unless they are for similar things.

Reply:
I see, so this is based on the on-chain [`VoteDirection`](https://github.com/yaadlabs/DAO/blob/d132454742be4a53d5d946a04fc76f6fd3c047c2/types/ApplicationTypes/Vote.lbf#L7) type, I kept the constructor names the same as they were originally.